pipx/commands.py

```diff
@@ -453,10 +453,22 @@ def uninstall(venv_dir: Path, package: str, local_bin_dir: Path, verbose: bool):
 
     venv = Venv(venv_dir, verbose=verbose)
 
-    metadata = venv.get_venv_metadata_for_package(package)
-    app_paths = metadata.app_paths
-    for dep_paths in metadata.app_paths_of_dependencies.values():
-        app_paths += dep_paths
+    if venv.python_path.is_file():
+        # has a valid python interpreter and can get metadata about the package
+        metadata = venv.get_venv_metadata_for_package(package)
+        app_paths = metadata.app_paths
+        for dep_paths in metadata.app_paths_of_dependencies.values():
+            app_paths += dep_paths
+    else:
+        # doesn't have a valid python interpreter. We'll take our best guess on what to uninstall
+        # here.
```

Comment:
The heuristic here is any symlink in `~/.local/bin` pointing to `.local/pipx/venvs/PACKAGE/bin`, which seems pretty reasonable, if not better than the default behavior.

Reply:
I think this is a reasonable approach 👍 Probably will have a non heuristic answer once pipxrc becomes a thing.